core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -241,36 +245,86 @@ FuturePtr<cosmos::TransactionsBulk> GaiaCosmosLikeBlockchainExplorer::getTransac
                 for (const auto &node : transactions) {
                     auto tx = cosmos::Transaction();
                     parseTransactionWithPosttreatment(node, tx);
-                    result->transactions.emplace_back(tx);
+                    result.transactions.emplace_back(tx);
                 }
 
                 if (!document.HasMember(kCount) ||
                     !document[kCount].IsString() ||
                     !document.HasMember(kTotalCount) ||
                     !document[kTotalCount].IsString()) {
-                    result->hasNext = false;
+                    result.hasNext = false;
                 }
                 else {
                     const auto count = std::stoi(document[cosmos::constants::kCount].GetString());
                     const auto total_count =
                         std::stoi(document[kTotalCount].GetString());
-                    result->hasNext = (count < total_count);
+                    result.hasNext = (count < total_count);
                 }
-                return result;
-            });
+                return Future<cosmos::TransactionsBulk>::successful(result);
+                })
+                .flatMapPtr<cosmos::TransactionsBulk>(
+                getContext(),
+                [this](const cosmos::TransactionsBulk& inputBulk) mutable -> FuturePtr<cosmos::TransactionsBulk> {
+                    auto retval = inputBulk;
+                    auto inputTxs = inputBulk.transactions;
+                    std::vector<FuturePtr<cosmos::Transaction>> filledTxsFutures;
+                    std::transform(
+                        inputTxs.cbegin(),
+                        inputTxs.cend(),
+                        std::back_inserter(filledTxsFutures),
+                        [this](const cosmos::Transaction &inputTx) {
+                            return this->inflateTransactionWithBlockData(inputTx);
+                        });
+                    return async::sequence(getContext(), filledTxsFutures)
+                        .flatMapPtr<cosmos::TransactionsBulk>(
+                            getContext(),
+                            [retval](
+                                const std::vector<std::shared_ptr<cosmos::Transaction>> &filledTxsList) mutable {
+                                retval.transactions.clear();
```

Comment:
The reallocation isn't guarantee with `clear` method. If you really need to free some memory space you can use swap method with an empty vector.

Reply:
It's not really for the memory space that I do it ; I'm not sure I can actually use `transform` to mutate the content of `retval/inputBulk.transactions`, so instead I just clear the vector and use the back_inserter. I don't know if there's a better way to do it, maybe with a swap in the transform function

Reply:
So you can do something like that :
```
retval.transactions.erase(
   std::transform(filledTxsList.cbegin(), filledTxsList.cend(), retval.transactions.begin(), ...),
   retval.transactions.end());
```

Reply:
Oh wait ! Since `filledTxsList` has the same size that `retval.transactions` - you construct the first container with the second one - you can skip the `erase` call which is result to a no operation.

Reply:
Seems good :)